requester/tracer.go

```diff
@@ -16,70 +16,86 @@ type Event struct {
 // tracer is a http.RoundTripper to be used as a http.Transport
 // that records the events of an outgoing HTTP request.
 type tracer struct {
-	start  time.Time
-	events []Event
+	start     time.Time
+	events    []Event
+	transport http.RoundTripper
 }
 
 // RoundTrip implements http.RoundTripper. It attaches the client trace
-// to the request context and calls http.DefaultTransport.RoundTrip
-// with the new created request.
-func (p *tracer) RoundTrip(r *http.Request) (*http.Response, error) {
-	ctx := httptrace.WithClientTrace(r.Context(), p.trace())
-	return http.DefaultTransport.RoundTrip(r.WithContext(ctx))
+// to the request context and calls t.transport.RoundTrip with a client
+// trace attached to the request context.
+func (t *tracer) RoundTrip(r *http.Request) (*http.Response, error) {
+	ctx := httptrace.WithClientTrace(r.Context(), t.trace())
+	return t.transport.RoundTrip(r.WithContext(ctx))
+}
+
+// CloseIdleConnections closes any connections on its http.Transport
+// which are sitting idle in a "keep-alive" state.
+//
+// If the tracer's Transport does not have a CloseIdleConnections method
+// then this method does nothing.
```

Comment:
More doc, as it is not obvious ?
```suggestion
// CloseIdleConnections closes any connections on its http.Transport
// which are sitting idle in a "keep-alive" state.
//
// It provides access to the Tracer private Transport's CloseIdleConnections
// method. If the Tracer's Transport does not have a CloseIdleConnections
// method then this method does nothing.
```